modules/programs/polybar.nix

```diff
@@ -0,0 +1,98 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.polybar;
+
+in
+
+{
+  options = {
+    programs.polybar = {
+      enable = mkEnableOption "Polybar status bar";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.polybar;
+        defaultText = "pkgs.polybar";
+        description = "Polybar package to install.";
+        example =  literalExample ''
+          (pkgs.polybar.override {
+            i3GapsSupport = true;
+            alsaSupport = true;
+            iwSupport = true;
+            githubSupport = true;
+          })
+        '';
+      };
+
+      config = mkOption {
+        type = types.either (types.attrsOf types.attrs) types.path;
+        description = ''
+          Polybar configuration. Can be either path to a file, or set of attibutes
+          that will be used to create the final configuration.
+        '';
+        default = {};
+        example = literalExample ''
+          config = {
+            "bar/top" = {
+              monitor = "\''${env:MONITOR:eDP1}";
+              width = "100%";
+              height = "3%";
+              radius = 0;
+              modules-center = "date";
+            };
+
+            "module/date" = {
+              type = "internal/date";
+              internal = 5;
+              date = "%d.%m.%y";
+              time = "%H:%M";
+              label = "%time%  %date%";
+            };
+          };
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        description = "Additional configuration to add.";
+        default = "";
+        example = literalExample ''
+          extraConfig = '''
+            [module/date]
+            type = internal/date
+            interval = 5
+            date = "%d.%m.%y"
+            time = %H:%M
+            format-prefix-foreground = \''${colors.foreground-alt}
+            label = %time%  %date%
+          ''';
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable (
+    mkMerge [
+      {
+        home.packages = [ cfg.package ];
+      }
+
+      (mkIf (lib.isAttrs cfg.config) {
+        home.file.".config/polybar/config".text =
+          generators.toINI {} cfg.config;
+      })
+
+      (mkIf (!lib.isAttrs cfg.config) {
+        home.file.".config/polybar/config".source = cfg.config;
+      })
+
+      {
+        home.file.".gitconfig".text = cfg.extraConfig;
```

Comment:
Huh? I'm a bit surprised to see `.gitconfig` here :-)

I'm guessing the extra config should be appended `.config/polybar/config"` but that seems tricky to me in the case where a user configures `programs.polybar.config` to be a path. You'd have to do something like

```
home.file.".config/polybar/config".text = readFile cfg.config + "\n" + cfg.extraConfig;
```

Looking at the Polybar wiki I saw the "[include-file](https://github.com/jaagr/polybar/wiki/Configuration#file-inclusion)" option. Perhaps an alternative is to set the type of `config` to

```
types.coercedTo
  types.path
  (p: { "section/base" = { include-file = p; }; })
  (types.attrsOf types.attrs);
```

Of course, I haven't tested this idea in the slightest ;-) But if it works then you shouldn't have problems with needing to identify whether `config` is an attribute set or a path and I guess the module's `config` section could be just

```
  config = mkIf cfg.enable {
    home.packages = [ cfg.package ];
    home.file.".config/polybar/config".text =
      generators.toINI {} cfg.config + "\n" + cfg.extraConfig;
  };
```